src/test/java/org/jenkinsci/plugins/workflow/utility/steps/zip/UnZipStepTest.java

```diff
@@ -0,0 +1,201 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.workflow.utility.steps.zip;
+
+import hudson.model.Label;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.steps.StepConfigTester;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.io.File;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.fail;
+
+/**
+ * Tests for {@link UnZipStep}.
+ *
+ * @author Robert Sandell &lt;rsandell@cloudbees.com&gt;.
+ */
+public class UnZipStepTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Before
+    public void setup() throws Exception {
+        j.createOnlineSlave(Label.get("slaves"));
+    }
+
+    @Test
+    public void configRoundTrip() throws Exception {
+        UnZipStep step = new UnZipStep("target/my.zip");
+        step.setDir("base/");
+        step.setGlob("**/*.zip");
+        step.setRead(true);
+
+        UnZipStep step2 = new StepConfigTester(j).configRoundTrip(step);
+        j.assertEqualDataBoundBeans(step, step2);
+        assertFalse(step2.isTest());
+    }
+
+    @Test
+    public void simpleUnZip() throws Exception {
+        WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(
+                "node('slaves') {\n" +
+                        "  dir('zipIt') {\n" +
+                        "    writeFile file: 'hello.txt', text: 'Hello World!'\n" +
+                        "    writeFile file: 'hello.dat', text: 'Hello World!'\n" +
+                        "    dir('two') {\n" +
+                        "      writeFile file: 'hello.txt', text: 'Hello World2!'\n" +
+                        "    }\n" +
+                        "    zip zipFile: '../hello.zip'\n" +
+                        "  }\n" +
+                        "  dir('unzip') {\n" +
+                        "    unzip '../hello.zip'\n" +
+                        "    String txt = readFile 'zipIt/hello.txt'\n" +
```

Comment:
Yes, I don't like it as well, but it's the way that `FilePath.zip` behaves.
I have been investigating how to change that behaviour but from what I can gather it would mean implementing it fully and not reusing anything.